From akpm@osdl.org Thu Jan  5 22:54:22 2006
From: David Hardeman <david@2gen.com>
Message-Id: <200601060650.k066opS4008827@shell0.pdx.osdl.net>
Subject: USB: iomega umini is unusual
To: greg@kroah.com
Cc: akpm@osdl.org, david@2gen.com
Date: Thu, 05 Jan 2006 22:50:38 -0800


From: David Hardeman <david@2gen.com>

The patch removes some harmless warning messages from generated by the
following Iomega usb key:

> usb-storage: waiting for device to settle before scanning
>   Vendor: I0MEGA    Model: UMni1GB*IOM2K4    Rev: 1.01
>   Type:   Direct-Access                      ANSI SCSI revision: 02
> SCSI device sda: 2048000 512-byte hdwr sectors (1049 MB)
> sda: Write Protect is off
> sda: Mode Sense: 00 00 00 00
> sda: assuming drive cache: write through

Warning message:

> ioctl_internal_command: <8 0 0 0> return code =3D 8000002
>    : Current: sense key=3D0x0
>     ASC=3D0x0 ASCQ=3D0x0
> SCSI device sda: 2048000 512-byte hdwr sectors (1049 MB)

Signed-off-by: David Hardeman <david@2gen.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/storage/unusual_devs.h |    9 +++++++++
 1 file changed, 9 insertions(+)

--- gregkh-2.6.orig/drivers/usb/storage/unusual_devs.h
+++ gregkh-2.6/drivers/usb/storage/unusual_devs.h
@@ -1154,6 +1154,15 @@ UNUSUAL_DEV(  0x4146, 0xba01, 0x0100, 0x
 		"Micro Mini 1GB",
 		US_SC_DEVICE, US_PR_DEVICE, NULL, US_FL_NOT_LOCKABLE ),
 
+/*
+ * David Hardeman <david@2gen.com>
+ * The key makes the SCSI stack print confusing (but harmless) messages
+ */
+UNUSUAL_DEV(  0x4146, 0xba01, 0x0100, 0x0100,
+		"Iomega",
+		"Micro Mini 1GB",
+		US_SC_DEVICE, US_PR_DEVICE, NULL, US_FL_NOT_LOCKABLE ),
+
 #ifdef CONFIG_USB_STORAGE_SDDR55
 UNUSUAL_DEV(  0x55aa, 0xa103, 0x0000, 0x9999, 
 		"Sandisk",
